processor/src/test/java/org/mapstruct/ap/test/nestedbeans/RecursionTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.nestedbeans;
+
+import java.util.Collections;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.test.nestedbeans.recursive.RecursionMapper;
+import org.mapstruct.ap.test.nestedbeans.recursive.TreeRecursionMapper;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+@RunWith(AnnotationProcessorTestRunner.class)
+public class RecursionTest {
+
+    @WithClasses({
+        RecursionMapper.class
+    })
+    @Test
+    @IssueKey("1103")
+    public void testRecursiveAutoMap() {
+        RecursionMapper.Root root = RecursionMapper.INSTANCE.mapRoot(
+            new RecursionMapper.RootDto(
+                new RecursionMapper.ChildDto( "Sub Root", new RecursionMapper.ChildDto(
+                    "Sub child", null
+                ) )
+            ) );
+
+        Assert.assertEquals(
```

Comment:
Can you use object accessors instead of the equals method for the checks? You can have a look at the latest `NestedSimpleBeansMappingTest`.

Then you won't need the `equals` and `hashCode` methods in the objects and the class will be smaller :)